rotkehlchen/chain/evm/decoding/weth/decoder.py

```diff
@@ -47,10 +54,18 @@ def __init__(
         self.counterparty = counterparty
 
     def _decode_wrapper(self, context: DecoderContext) -> DecodingOutput:
-        if context.tx_log.topics[0] == WETH_DEPOSIT_TOPIC:
+        input_data = context.transaction.input_data
+        # WETH on Arbitrum is deployed as proxy, we need to check the method
+        if context.tx_log.topics[0] == WETH_DEPOSIT_TOPIC or (
```

Comment:
no no no. Don't do it like this. That's going against inheritance. Add extra logic in arbitrum's decoder class.